src/test/java/org/opensearch/security/test/AbstractSecurityUnitTest.java

```diff
@@ -194,6 +197,28 @@ public void waitForInit(Client client) {
         }
     }
 
+    public static Settings.Builder nodeRolesSettings(final Settings.Builder settingsBuilder, final boolean isClusterManager, final boolean isDataNode) {
+        final ImmutableList.Builder<String> nodeRolesBuilder = ImmutableList.<String>builder();
+        if (isDataNode) {
+            nodeRolesBuilder.add("data");
+        }
+        if (isClusterManager) {
+            nodeRolesBuilder.add("cluster_manager");
+        }
+
+        final Settings nodeRoleSettings = Settings.builder().putList(NODE_ROLE_KEY, nodeRolesBuilder.build()).build();
+        return mergeNodeRolesAndSettings(settingsBuilder, nodeRoleSettings);
+    }
+
+    public static Settings.Builder mergeNodeRolesAndSettings(final Settings.Builder settingsBuilder, final Settings otherSettings) {
```

Comment:
Why add this method?

Settings.Builder.put(Settings) [[code](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/common/settings/Settings.java#L1032)] takes the map representation of the incoming settings and stomps over top _all_ intersecting keys.

Depending when `.put(settings)` was used this would stomp the `{"node.roles": ["cluster_manager"]}` -> `{"node.roles":["remote_cluster_client"]}` - this would stop the cluster from starting because it couldn't find any cluster_manager eligable nodes OR the inverse would happen and `remote_cluster_client` role would not be available on the cluster.

^ I started that sentence with _depending_, and yes that is because sometimes the test configuration is providing the default values before the cluster helper does its thing, and sometimes it's provided through a `NodeSettingsSupplier` which is applied afterward, but before the cluster starts up.

This method is available to handle the bi-directional merges of these settings and ensure the `node.roles` are merged instead of replaced.

### That was difficult and complex, why go through that trouble for only `node.roles`?
Why not just support something like `.merge(Settings)`? - IMO this is a better solution except of course if our cluster configuration is taking advantage of this functionality - its a feature not a bug!

I avoided tackling this problem because the data flow around these settings is non-trivial and debugging and troubleshooting hundreds of integration unit tests compared to fixing the smaller subset of cross cluster tests that are broken seemed like the better short-term path.

As we build on the new test framework, being rigorous around the data flow can help avoid this class of problems

Reply:
Thanks for the explanation for this method. 

Would making this more generic and having the method take in a key be useful?

i.e. `public static Settings.Builder mergeListSettingByKey(final Settings.Builder settingsBuilder, final Settings otherSettings, String settingKey)`

Reply:
^ Next time we need to do this I think it would be worthwhile.

Reply:
@peternied Thanks for the comments making review much easier!

>Would making this more generic and having the method take in a key be useful?

@cwperks very interesting idea! I might miss something. Are there any other use cases where `settingKey` can be something else other than `NODE_ROLE_KEY`? If so, making the method more generic would be necessary to serve those use cases.

Reply:
Interesting..thank you for the explanation. I like the idea of making it generic if there are other values as `List` objects that would "stomp" each other

Reply:
We do not have any other use cases at the moment.  If we found ourselves using a function like this often I think it would be more worthwhile to make a pull request to update Settings.Builder with this functionality, but that seems out of scope for this PR.  What do you think @cwperks @cliu123 ?

Reply:
If there are not other use cases, then I am good with moving forward as-is. @cwperks What do you think?

Reply:
@cliu123 I believe anywhere where you see `putList` in the code would be eligible: https://github.com/opensearch-project/security/search?q=putList